pkg/export/disk/disk.go

```diff
@@ -0,0 +1,271 @@
+package disk
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/export/util"
+	"k8s.io/client-go/util/retry"
+)
+
+type Connection struct {
+	// path to store audit logs
+	Path string `json:"path,omitempty"`
+	// max number of audit results to store
+	MaxAuditResults int `json:"maxAuditResults,omitempty"`
+	// File to write audit logs
+	File *os.File
+
+	// current audit run file name
+	currentAuditRun string
+}
+
+type Writer struct {
+	openConnections map[string]Connection
+}
+
+const (
+	maxAllowedAuditRuns = 5
+)
+
+const (
+	Name = "disk"
+)
+
+var Connections = &Writer{
+	openConnections: make(map[string]Connection),
+}
+
+func (r *Writer) CreateConnection(_ context.Context, connectionName string, config interface{}) error {
+	cfg, ok := config.(map[string]interface{})
+	if !ok {
+		return fmt.Errorf("invalid config format")
+	}
+
+	path, pathOk := cfg["path"].(string)
+	if !pathOk {
+		return fmt.Errorf("missing or invalid values in config for connection: %s", connectionName)
+	}
+	var err error
+	maxResults, maxResultsOk := cfg["maxAuditResults"].(float64)
```

Comment:
Why float?

Reply:
Because the inherent value is stored in `float64` when the object is getting unpacked, so checking and converting to `int64` does not work here.